test/testrunner.py

```diff
@@ -38,21 +43,137 @@ def run_sagemaker_local_tests(images):
+        context = Context()
+        with context.cd(path):
+            context.run(f"python3 -m virtualenv {tag}")
+            with context.prefix(f"source {tag}/bin/activate"):
+                context.run("pip install -r requirements.txt", warn=True)
+                context.run(pytest_command)
+    except Exception as e:
+        LOGGER.error(e)
+        return False
+
+    return True
+
+
+def print_log_stream(logs):
+    """
+    print the log stream from Job Executor
+    :param logs: <dict> the returned dict from JobRequester.receive_logs
+    """
+    LOGGER.info("Log stream from Job Executor.....")
+    print(logs["LOG_STREAM"])
+    LOGGER.info("Print log stream complete.")
+
+
+def send_scheduler_requests(requester, image):
+    """
+    Send a PR test request through the requester, and wait for the response.
+    If test completed or encountered runtime error, create local XML reports.
+    Otherwise the test failed, print the failure reason.
+
+    :param requester: JobRequester object
+    :param image: <string> ECR URI
+    """
+    # Note: 3 is the max number of instances required for any tests. Here we schedule tests conservatively.
+    identifier = requester.send_request(image, "PR", 3)
+    image_tag = image.split(":")[-1]
+    report_path = os.path.join(os.getcwd(), "test", f"{image_tag}.xml")
+    while True:
+        query_status_response = requester.query_status(identifier)
+        test_status = query_status_response["status"]
+        if test_status == "completed":
+            LOGGER.info(f"Test for image {image} completed.")
+            logs_response = requester.receive_logs(identifier)
+            LOGGER.info(f"Receive logs success for ticket {identifier.ticket_name}, report path: {report_path}")
+            print_log_stream(logs_response)
+            metrics_utils.send_test_result_metrics(0)
+            with open(report_path, "w") as xml_report:
+                xml_report.write(logs_response["XML_REPORT"])
+            break
+
+        elif test_status == "runtimeError":
+            LOGGER.warning(f"Tests for image {image} ran into runtime error.")
+            logs_response = requester.receive_logs(identifier)
+            with open(report_path, "w") as xml_report:
+                xml_report.write(logs_response["XML_REPORT"])
+            print_log_stream(logs_response)
+            metrics_utils.send_test_result_metrics(1)
+            break
+
+        elif test_status == "failed":
+            LOGGER.warning(f"Scheduling failed. Reason: {query_status_response['reason']}")
+            metrics_utils.send_test_result_metrics(1)
+            break
+
+
 def run_sagemaker_remote_tests(images):
     """
     Function to set up multiprocessing for SageMaker tests
     :param images: <list> List of all images to be used in SageMaker tests
     """
-    if not images:
+    use_scheduler = os.getenv("USE_SCHEDULER", "False")
+    executor_mode = os.getenv("EXECUTOR_MODE", "False")
```

Comment:
nit: You can make these variables boolean by comparing the values you get for the env variables to "true".

So it would look like:
```
use_scheduler = os.getenv("USE_SCHEDULER", "False").lower() == "true"
executor_mode = os.getenv("EXECUTOR_MODE", "False").lower() == "true"
```
This way, when you need  to use these variables in if-else conditions, you don't have to compare them to "true" there every time. Especially since these variables are supposed to be representing boolean True/False anyway.